roofit/roofit/test/testRooBernstein.cxx

```diff
@@ -0,0 +1,115 @@
+// Test for RooBernstein
+// Authors: Rahul Balasubramanian, CERN  10/2019
+
+#include "RooRealVar.h"
+#include "RooDataSet.h"
+#include "RooExtendPdf.h"
+#include "RooBernstein.h"
+#include "TCanvas.h"
+#include "TMath.h"
+#include "RooPlot.h"
+#include "TAxis.h"
+
+#include "gtest/gtest.h"
+
+using namespace RooFit;
+
+Double_t integral(double t) {
+  Double_t val = 0.3*t + 6*1e-3*TMath::Power(t,2) - 9*1e-5*TMath::Power(t,3) + 4.25*1e-7*TMath::Power(t,4);
+  return val;
+};
+
+void runFit(unsigned int N, double a0, double a1, double a2, double a3)
+{
+  RooRealVar x("x", "x", 0., 100.);
+
+  // Define coefficients for a bernstein polynomial of order 3
+  RooRealVar c0("c0", "c1 coeff", a0, 0., 10.);
+  RooRealVar c1("c1", "c1 coeff", a1, 0., 10.);
+  RooRealVar c2("c2", "c2 coeff", a2, 0., 10.);
+  RooRealVar c3("c3", "c3 coeff", a3, 0., 10.);
+
+  // Build bernstein p.d.f in terms of coefficients
+  RooBernstein bern("bern", "bernstein PDF", x, RooArgList(c0, c1, c2, c3));
+
+  // Set ranges for the variable
+  x.setRange("range1", 0., 30.);
+  x.setRange("range2", 30., 70.);
+  x.setRange("range3", 70., 100.);
+  x.setRange("FULL", 0., 100.);
+
+  // Set normalization range
+  bern.selectNormalizationRange("FULL",kTRUE);
+
+  // Test ranged integration
+  Double_t int_range1 = bern.analyticalIntegral(1, "range1");
+  Double_t int_range2 = bern.analyticalIntegral(1, "range2");
+  Double_t int_range3 = bern.analyticalIntegral(1, "range3");
+  Double_t int_full = bern.analyticalIntegral(1, "FULL");
+
+
+  // closure
+  EXPECT_LT(fabs(int_full - int_range1 - int_range2 - int_range3), 1e-10);
+
+  // comparision with polynomial
+  EXPECT_LT(fabs(int_range1 - integral(30.) - integral(0.)),1e-3);
+  EXPECT_LT(fabs(int_range2 - integral(70.) - integral(30.)),1e-3);
+  EXPECT_LT(fabs(int_range3 - integral(100.) - integral(70.)),1e-3);
+  EXPECT_LT(fabs(int_full - integral(100.) - integral(0.)),1e-3);
+
+
+  // Create an extended pdf to fit simultaneously in the two ranges 
+  RooRealVar Ne("Ne", "number of events", 1000, -1e30, 1e30);
+  RooExtendPdf extbern("extbern", "bernstein extended pdf", bern, Ne, "FULL");
+
+  // Create a dataset from the bernstein pdf
+  RooDataSet *data = bern.generate(x, N);
+
+  // Fit pdf to data
+  c0.setConstant(kTRUE);
+  auto result = extbern.fitTo(*data,RooFit::Range("range1,range2"));
```

Comment:
```suggestion
  extbern.fitTo(*data,RooFit::Range("range1,range2"));
```
The result will only be saved if you use `Save()`, so ignoring the result will fix the warning.